### PR TITLE
todo task for GCI - moving location_id to Post linked table

### PR DESCRIPTION
**Pull request by samsruti**
Link to Todo : https://github.com/flavour/eden/blob/master/modules/s3db/cms.py#L190
Link To GCI task : https://www.google-melange.com/gci/task/view/google/gci2014/5910927769600000

### CODE REVIEW

File: static/themes/default/widgets.css [hunk 1]
```diff
@@ -560,7 +560,13 @@ ul.ui-autocomplete {
     color: red;
     font-weight: bold;
 }
-
+.top-bar-section .right li .dropdown {
```
Comment (nursix) [inline static/themes/default/widgets.css]: -1 Top bar is controlled by foundation framework, should not be overwritten in widget.css

File: static/styles/foundation/foundation.css [hunk 2]
```diff
@@ -6381,7 +6381,6 @@ meta.foundation-mq-topbar {
     position: static !important;
     height: 100px;
     width: auto;
-    overflow: scroll;
```
Comment (nursix) [inline static/styles/foundation/foundation.css:6384]: -1 This file is based on foundation framework, please commit upstream

File: static/styles/foundation/foundation.css [hunk 3]
```diff
@@ -6459,8 +6458,6 @@ meta.foundation-mq-topbar {
   .top-bar-section .right li .dropdown {
     left: auto;
     right: 0;
-    max-height: 350px;
-    overflow-y: auto;
```
Comment (nursix) [inline static/styles/foundation/foundation.css:6463]: -1 This file should not be modified - please commit upstream or customize SCSS.

File: static/themes/default/widgets.css (hunk 1 above)
Comment (nursix) [inline static/themes/default/widgets.css]: What's the point with this change, anyway?

File: modules/s3db/cms.py [hunk 4]
```diff
@@ -240,6 +238,45 @@ def model(self):
             msg_record_deleted = T("Post deleted"),
             msg_list_empty = T("No posts currently available"))
 
+        crud_form = S3SQLCustomForm("name",
+                                    "title",
+                                    "body",
+                                    S3SQLInlineComponent ("post_location", 
+                                            label = T("Location"),
```
Comment (nursix) [inline modules/s3db/cms.py]: Bad indent

File: modules/s3db/cms.py [hunk 5]
```diff
@@ -240,6 +238,45 @@ def model(self):
...
+                                            label = T("Location"),      
+                                            field = "location_id",                                   
+                                            ),
+                                    "avatar",
+                                    "replies",
+                                    "expired",
+                                    "comments",
+                                    )        
+        self.add_components(tablename,
+                       cms_post_location = {"link": "cms_post_location",
```
Comment (nursix) [inline modules/s3db/cms.py]: Bad indent

File: modules/s3db/cms.py [hunk 6]
```diff
@@ -240,6 +238,45 @@ def model(self):
...
+        self.configure(tablename,
+                       crud_form=crud_form)        
+        # ---------------------------------------------------------------------
+        # Post Locations (link table)
+        #
+        tablename = "cms_post_location"
+        define_table(tablename,
+                     series_id(),
+                     self.gis_location_id(
+                        requires = IS_LOCATION(),
```
Comment (nursix) [inline modules/s3db/cms.py]: Not required, no.

File: modules/s3db/cms.py [hunk 7]
```diff
@@ -240,6 +238,45 @@ def model(self):
...
+                        represent = current.s3db.gis_location_represent,
```
Comment (nursix) [inline modules/s3db/cms.py]: default

File: modules/s3db/cms.py [hunk 8]
```diff
@@ -240,6 +238,45 @@ def model(self):
...
+                        widget = S3LocationAutocompleteWidget(level=current.gis.hierarchy_level_keys),
```
Comment (nursix) [inline modules/s3db/cms.py]: Err - what? Doesn't make sense to me - first you want a location selector, then you configure an autocomplete?

File: static/themes/default/widgets.css [hunk 9]
```diff
@@ -560,7 +560,13 @@ ul.ui-autocomplete {
...
+    left: auto;
+    right: 0;
+    max-height: 360px;
```
Comment (nursix) [inline static/themes/default/widgets.css]: Definitely not max-height!

File: static/themes/default/widgets.css [hunk 10]
```diff
@@ -560,7 +560,13 @@ ul.ui-autocomplete {
     color: red;
     font-weight: bold;
 }
-
+.top-bar-section .right li .dropdown {
+    left: auto;
+    right: 0;
```
Comment (nursix) [inline static/themes/default/widgets.css]: Can't do that - invalid with RTL languages

**Comment by nursix** [comment]
I would recommend to take out the CSS changes, fix the indent, and re-consider the location auto-complete (why should this be an autocomplete?).

File: modules/s3db/cms.py (hunk 5 above)
Comment (samsruti) [inline modules/s3db/cms.py]: Now corrected

**Comment by samsruti** [comment]
Css changes is working properly for LTR and RTL direction in all templates. Default template has now scrollable dropdown menu .

**Comment by nursix** [comment]
Yeah - I hear you, but you should however not change foundation.css nor put the "fix" into widget.css (does simply not belong there). And in fact, I see this as a non-issue, so I wouldn't merge it as it looks a bit clumsy and doesn't adapt to the screen size anyway - so it's not really a "solution" but just sort-of a band-aid.

NB Mobile browsers are more important to test, and here it's not really a useful change.

**Comment by samsruti** [comment]
ok got it sir . so can i proceed to another bug ?I want to work on this : http://sunflower.sahanafoundation.org/eden/project/task/361

**Comment by nursix** [comment]
You can do whatever you like ;)

However, I don't see the "bug" in that task - what's wrong with the pie chart?

**Comment by samsruti** [comment]
The pie chart is not displaying on the basis of person by age group

**Comment by nursix** [comment]
Yeah - /if/ we would use the "age group" field - which we do not use in default. So it's never populated, that's why it doesn't show anything.

However, these charts have no practical meaning - they were a PoC for dashboard-style module homepages a long, long time ago. I would just remove the charts - they are not useful in any way (even if they did work correctly).

So many students tried again and again to "fix" them - but there is really nothing to fix, they are "just a test" on a normally hidden page (no regular user would see the page in normal deployments, only admins, and they wouldn't use it for anything).

But if you have time to waste - please...

**Comment by samsruti** [comment]
Heheheh... No i dont want to do that :P . @nursix  can you suggest some bugs which will be very useful . I would like to work on those task :+1:

**Comment by ramdesh** [comment]
Merging from @samsruti for GCI Task https://www.google-melange.com/gci/task/view/google/gci2014/5910927769600000

**Comment by ramdesh** [comment]
@nursix : Note that only changes in cms remain. CSS fixes etc have been removed.

**Pull request merged**